src/app/clusters/scenes/SceneTableImpl.cpp

```diff
@@ -0,0 +1,680 @@
+{
+    if (!this->HandlerListEmpty())
+    {
+        for (uint8_t i = 0; i < this->mNumHandlers; i++)
+        {
+            clusterId cArray[kMaxClusterPerScenes];
+            Span<clusterId> cSpan(cArray);
+            if (this->mHandlers[i] != nullptr)
+            {
+                this->mHandlers[i]->GetSupportedClusters(scene.mStorageId.mEndpointId, cSpan);
+                for (uint8_t j = 0; j < cSpan.size(); j++)
+                {
+                    ExtensionFieldsSet EFS;
+                    MutableByteSpan EFSSpan = MutableByteSpan(EFS.mBytesBuffer, kMaxFieldsPerCluster);
+
+                    EFS.mID = cArray[j];
+                    ReturnErrorOnFailure(this->mHandlers[i]->SerializeSave(scene.mStorageId.mEndpointId, EFS.mID, EFSSpan));
+                    EFS.mUsedBytes = (uint8_t) EFSSpan.size();
+                    ReturnErrorOnFailure(scene.mStorageData.mExtensionFieldSets.InsertFieldSet(EFS));
+                }
+            }
+        }
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR DefaultSceneTableImpl::SceneApplyEFS(FabricIndex fabric_index, const SceneStorageId & scene_id)
+{
+    FabricSceneData fabric(fabric_index);
+    SceneTableData scene(fabric_index);
+    ExtensionFieldsSet EFS;
+    TransitionTimeMs time;
+    clusterId cluster;
+
+    ReturnErrorOnFailure(fabric.Load(mStorage));
+    VerifyOrReturnError(fabric.Find(scene_id, scene.index) == CHIP_NO_ERROR, CHIP_ERROR_NOT_FOUND);
+    ReturnErrorOnFailure(scene.Load(mStorage));
+
+    if (!this->HandlerListEmpty())
+    {
+        for (uint8_t i = 0; i < scene.mStorageData.mExtensionFieldSets.GetFieldNum(); i++)
+        {
+            scene.mStorageData.mExtensionFieldSets.GetFieldSetAtPosition(EFS, i);
+            cluster = EFS.mID;
+            time    = scene.mStorageData.mSceneTransitionTimeSeconds * 1000 +
+                (scene.mStorageData.mTransitionTime100ms ? scene.mStorageData.mTransitionTime100ms * 10 : 0);
+            ByteSpan EFSSpan = MutableByteSpan(EFS.mBytesBuffer, EFS.mUsedBytes);
+
+            if (!EFS.IsEmpty())
+            {
+                for (uint8_t j = 0; j < this->mNumHandlers; j++)
+                {
+                    ReturnErrorOnFailure(this->mHandlers[j]->ApplyScene(scene.mStorageId.mEndpointId, cluster, EFSSpan, time));
+                }
+            }
+        }
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR DefaultSceneTableImpl::RemoveFabric(FabricIndex fabric_index)
+{
+    FabricSceneData fabric(fabric_index);
+    SceneIndex idx = 0;
+    CHIP_ERROR err = fabric.Load(mStorage);
+    VerifyOrReturnError(CHIP_NO_ERROR == err || CHIP_ERROR_NOT_FOUND == err, err);
+
+    while (idx < kMaxScenePerFabric)
+    {
+        err = RemoveSceneTableEntryAtPosition(fabric_index, idx);
+        VerifyOrReturnError(CHIP_NO_ERROR == err || CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND == err, err);
+        idx++;
+    }
+
+    // Remove fabric
+    return fabric.Delete(mStorage);
```

Comment:
Again, how is this interacting with groups?  As far as I can tell, "badly".

At the very least, we need some tests that add some groups and scenes, remove the scenes, and check that the groups still work.